app/[locale]/(interactive)/4-17/page.tsx

```diff
@@ -0,0 +1,148 @@
+"use client";
+import ConfirmPopup from "@/components/ui/confirmPopup";
+import DownloadImageAlert from "@/components/ui/downloadImageAlert";
+import NextButton from "@/components/ui/nextButton";
+import { useNameStorkeStore } from "@/stores/NameStroke.store";
+import { useShareYourselfWordsStore } from "@/stores/ShareYourselfWords.store";
+import { useStickerStore } from "@/stores/sticker.store";
+import { toPng } from "html-to-image";
+import { useTranslations } from "next-intl";
+import Image from "next/image";
+import { getStroke } from "perfect-freehand";
+import { useCallback, useRef, useState } from "react";
+
+type Props = {};
+
+const Page = (props: Props) => {
+  const t = useTranslations("4-17");
+
+  const words = useShareYourselfWordsStore(
+    (state: { words: string }) => state.words,
+  );
+  const stickerBgPath = useStickerStore(
+    (state: { stickerBgPath: string }) => state.stickerBgPath,
+  );
+  const wordsRef = useRef<HTMLDivElement>(null);
+  const nameStorke = useNameStorkeStore(
+    (state: { nameStorke: number[][][] }) => state.nameStorke,
+  );
+  const [popup, setPopup] = useState<boolean>(false);
+  const [downloadAlert, setDownloadAlert] = useState<boolean>(false);
+
+  function getSvgPathFromStroke(stroke: number[][]) {
+    if (!stroke.length) return "";
+    // reduce the size of svg by divisor
+    const divisor = 5;
+    const d = stroke.reduce(
+      (acc: any[], [x0, y0]: any, i: number, arr: string | any[]) => {
+        const [x1, y1] = arr[(i + 1) % arr.length];
+        acc.push(
+          x0 / divisor,
+          y0 / divisor,
+          (x0 + x1) / 2 / divisor,
+          (y0 + y1) / 2 / divisor,
+        );
+        return acc;
+      },
+      ["M", stroke[0][0] / divisor, stroke[0][1] / divisor, "Q"],
+    );
+    d.push("Z");
+    return d.join(" ");
+  }
```

Comment:
This code is duplicated, consider moving to a place like `utils` so the function can be shared. Maybe `utils/svg.ts`